### PR TITLE
chore: add V0 migration guides

### PR DESCRIPTION
**Pull request by chpalac**
## Overview

Adding missing V0 migration guides

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/20221afa1c8dde9afe17a278ddcc0a8a9a4817e2'>20221afa1c8dde9afe17a278ddcc0a8a9a4817e2</a></sub>
<div id="bundleSizeComment9423"/>

**Review by jurokapsiar — APPROVED** [review]

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+<Meta title="Concepts/Migration/from v0/Components/Ref Migration" />
+
+# Ref component Migration guide
+
+The `Ref` component from V0 is deprecated as it used the deprecated `findDomNode` utility from React. All V9 components support ref forwarding, so you can simply use `useRef` to the component.
+
+Before:
+
+```tsx
+import { Flex, Flex } from '@fluentui/react-northstar';
```
Comment (jurokapsiar) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx:12]:
```suggestion
import { Button } from '@fluentui/react-northstar';
```

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const Component = ({ popupRef }) => (
```
Comment (jurokapsiar) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx:13]:
```suggestion
const Component = ({ buttonRef }) => (
```

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  <Ref innerRef={popupRef}>
```
Comment (jurokapsiar) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx:14]:
```suggestion
  <Ref innerRef={buttonRef}>
```

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <Flex>some content</Flex>
```
Comment (jurokapsiar) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx:15]:
```suggestion
    <Button content="some content" />
```

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  </Ref>
+);
+```
+
+After:
+
+```tsx
+import { Flex } from '@fluentui/react-components';
```
Comment (jurokapsiar) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx:23]:
```suggestion
import { Button } from '@fluentui/react-components';
```

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+const Component = () => {
+  const ref = React.useRef(null);
+
+  return <Flex ref={ref}>some content</Flex>;
```
Comment (jurokapsiar) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Ref.stories.mdx:27]:
```suggestion
  return <Button ref={ref}>some content</Button>;
```

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Flex.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    ...(isSomething && { color: 'red' }),
+  }),
+};
+```
+
+After:
+
+```tsx
+// in COMPONENT_NAME.tsx
+import { Flex } from '@fluentui/react-components';
```
Comment (jurokapsiar) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Flex.stories.mdx:87]: should we modify the package name?

Comment (chpalac) [reply apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Flex.stories.mdx]: Done, replaced by `@fluentui/react-migration-v0-v9`

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Button.stories.mdx [hunk 8]
```diff
@@ -6,6 +6,8 @@ import { Meta } from '@storybook/addon-docs';
 
 ## Overview:
 
+To help with the migration we also offer a mixin that [can be checked here](/?path=/docs/migration-shims-v0-buttonmixins--disabled-cursor)
```
Comment (ling1726) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Button.stories.mdx:9]: This link leads to a blank page for me

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Flex.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,113 @@
+import { Meta } from '@storybook/addon-docs';
```
Comment (ling1726) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Flex.stories.mdx:1]: This is really confusing, we have multiple Flex migration pages

image

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Grid.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+```tsx
+import { Grid } from '@fluentui/react-northstar';
+const Component = () => <Grid />;
+```
+
+After:
+
+```tsx
+import { Grid } from '@fluentui/react-components';
```
Comment (ling1726) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Grid.stories.mdx:21]: react-components does not export a `Grid` component

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/ItemLayout.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+```tsx
+import { ListItem } from '@fluentui/react-northstar';
+const Component = () => <ListItem />;
+```
+
+After:
+
+```tsx
+import { ItemLayout } from '@fluentui/react-components';
```
Comment (ling1726) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/ItemLayout.stories.mdx:26]: ditto, fix the import please

File: apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Segment.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+# Segment component Migration guide
+
+## Overview:
+
+To help with the migration we also offer a shim that [can be checked here](/?path=/docs/migration-shims-v0-segmentshim--default)
+
+Before:
+
+```tsx
+import { Segment, Button } from '@fluentui/react-northstar';
```
Comment (ling1726) [inline apps/public-docsite-v9/src/Concepts/Migration/FromV0/Components/Segment.stories.mdx:14]: v9 does not export a Segment component

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Avatar-mount_base.html">1283</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Avatar-mount.html">1289</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Button-mount_base.html">913</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Button-mount.html">919</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Field-mount_base.html">1959</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Field-mount.html">1929</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProvider-mount_base.html">1531</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProvider-mount.html">1533</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProviderWithTheme-mount_base.html">576</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProviderWithTheme-mount.html">583</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">536</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">542</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">585</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">575</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/InfoButton-mount_base.html">515</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/InfoButton-mount.html">514</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/MakeStyles-mount_base.html">1958</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/MakeStyles-mount.html">1953</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Persona-mount_base.html">2870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/Persona-mount.html">2799</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/SpinButton-mount_base.html">2259</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26118/perf-test-react-components/SpinButton-mount.html">2303</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by ling1726 — APPROVED** [review]

**Pull request merged**